yt_dlp/extractor/err.py

```diff
@@ -0,0 +1,203 @@
+        'url': 'https://jupiterpluss.err.ee/1609180445/bolee-zelenyj-tallinn',
+        'md5': '1b812270c4daf6ce51c06bfeaf33ed95',
+        'info_dict': {
+            'id': '1609180445',
+            'ext': 'mp4',
+            'title': 'Более зеленый Таллинн',
+            'alt_title': '',
+            'description': 'md5:fd34d9bf939c28c4a725b19a7f0d6320',
+            'release_date': '20231224',
+            'upload_date': '20231130',
+            'modified_date': '20231207',
+            'release_timestamp': 1703423400,
+            'timestamp': 1701338400,
+            'modified_timestamp': 1701967200,
+            'release_year': 2023,
+        },
+    }, {
+        'note': 'Jupiter+: Series: The Sniffer',
+        'url': 'https://jupiterpluss.err.ee/1608311387/njuhach',
+        'md5': '2abdeb7131ce551bce49e8d0cea08536',
+        'info_dict': {
+            'id': '1608311387',
+            'ext': 'mp4',
+            'title': 'Нюхач',
+            'alt_title': '',
+            'description': 'md5:8c5c7d8f32ec6e54cd498c9e59ca83bc',
+            'release_date': '20230601',
+            'upload_date': '20210818',
+            'modified_date': '20210903',
+            'release_timestamp': 1685633400,
+            'timestamp': 1629318000,
+            'modified_timestamp': 1630686000,
+            'release_year': 2013,
+            'episode': 'Episode 1',
+            'episode_id': '1608311390',
+            'episode_number': 1,
+            'season': 'Season 1',
+            'season_number': 1,
+            'series': 'Нюхач',
+            'series_id': '1608311387',
+        },
+    }, {
+        'note': 'Jupiter+: Podcast: lesnye-istorii-aisty',
+        'url': 'https://jupiterpluss.err.ee/1608990335/lesnye-istorii-aisty',
+        'md5': '8b46d7e4510b254a14b7a52211b5bf96',
+        'info_dict': {
+            'id': '1608990335',
+            'ext': 'm4a',
+            'title': 'Лесные истории | Аисты',
+            'alt_title': '',
+            'description': 'md5:065e721623e271e7a63e6540d409ca6b',
+            'release_date': '20230609',
+            'upload_date': '20230527',
+            'modified_date': '20230608',
+            'release_timestamp': 1686308700,
+            'timestamp': 1685145600,
+            'modified_timestamp': 1686252600,
+            'release_year': 2023,
+            'episode': 'Episode 0',
+            'episode_id': '1608990335',
+            'episode_number': 0,
+            'season': 'Season 0',
+            'season_number': 0,
+            'series': 'Лесные истории | Аисты',
+            'series_id': '1037497',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        data = try_call(lambda: self._download_json(
+            'https://services.err.ee/api/v2/vodContent/getContentPageData', video_id,
+            query={'contentId': video_id})['data']['mainContent'])
+        if not data:
+            raise ExtractorError('Failed to get video data', expected=True)
```

Comment:
```suggestion
        data = self._download_json(
            'https://services.err.ee/api/v2/vodContent/getContentPageData', video_id,
            query={'contentId': video_id})['data']['mainContent']
```

Reply:
previously you asked for error to be caught. but now TypeError or KeyError can happen if returned json is None or doesn't have either data or mainContent keys...

Reply:
I said it should be fatal. As in it should not be caught.
There is nothing logical we can do without the data, we don't need to prevent the error to then immediately throw a new error due no data.
We also don't expect the API to ever not return this data and if it does the extractor will most likely need to be updated.